lib/Driver/ToolChains.cpp

```diff
@@ -1287,6 +1287,26 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   if (!Linker.empty()) {
     Arguments.push_back(context.Args.MakeArgString("-fuse-ld=" + Linker));
   }
+  
+  // Configure the toolchain
+  // By default, use the system clang++ to link.
+  StringRef Clang = "clang++";
+  if (const Arg *A = context.Args.getLastArg(options::OPT_toolchain)) {
```

Comment:
Why wouldn't it work on Darwin?  Let's say you want to cross-compile for a Raspberry Pi from a Mac.  You definitely want to use the Linux armv7 or arm64 cross-linker.